packages/dev/core/src/Maths/math.vector.ts

```diff
@@ -5362,16 +5963,47 @@ export class Matrix {
      * Example Playground - https://playground.babylonjs.com/#AV9X17#49
      * @returns the matrix underlying array
      */
-    public toArray(): DeepImmutable<Float32Array | Array<number>> {
-        return this._m;
+    public toArray(array: FloatArray, index: number = 0): this {
+        const m = this._m;
+        for (let i = 0; i < m.length; i++) {
+            array[index + i] = m[i];
+        }
+        return this;
     }
+
     /**
      * Returns the matrix as a Float32Array or Array<number>
      * Example Playground - https://playground.babylonjs.com/#AV9X17#114
      * @returns the matrix underlying array.
      */
-    public asArray(): DeepImmutable<Float32Array | Array<number>> {
-        return this._m;
+    public asArray(): Tuple<number, 16> {
+        return [...this._m];

```

Comment:
This introduces an array copy here, could make a difference for performance.
![screenshot](https://github.com/BabylonJS/Babylon.js/assets/120431159/a63c555f-82f7-43d3-bfdc-dfae30780027)

Reply:
Patched `Matrix#asArray` for comparsion:
<https://playground.babylonjs.com/?snapshot=refs/pull/14235/merge#6HWS9M#57>
![screenshot](https://github.com/BabylonJS/Babylon.js/assets/120431159/013db5ac-c4f4-4f89-a76b-b9331f6c27e7)

Reply:
Fixed in a59d83444b